task_sdk/tests/api/test_client.py

```diff
@@ -133,3 +134,21 @@ def handle_request(request: httpx.Request) -> httpx.Response:
                 "reason": "not_found",
             }
         }
+
+    def test_variable_put_success(self):
+        # Simulate a successful response from the server when putting a variable
+        def handle_request(request: httpx.Request) -> httpx.Response:
+            if request.url.path == "/variables/test_key":
+                return httpx.Response(
+                    status_code=200,
+                    json={"message": "Variable successfully set"},
+                )
+            return httpx.Response(status_code=400, json={"detail": "Bad Request"})
+
+        client = make_client(transport=httpx.MockTransport(handle_request))
+
+        msg = PutVariable(
+            key="test_key", value="test_value", description="test_description", type="PutVariable"
```

Comment:
Should `value` always be JSON-encoded, in which case this might be more representative. 
```suggestion
            key="test_key", value='"test_value"', description="test_description", type="PutVariable"
```

Reply:
Not really, it can be str normally unlike with XCOMs. 
```
class VariablePostBody(BaseModel):
    """Request body schema for creating variables."""

    value: str | None = Field(serialization_alias="val")
    description: str | None = Field(default=None)
```